### PR TITLE
chore(ci_cd): added action to create release

### PR DESCRIPTION
**Pull request by laurentlp**
This PR adds an action ("Create Release Pull Request") whose task is to automate the creation of the release PR. It also allows us to bump the version of the root package.json while working in the next version without automatically releasing a new binary. 

Also closes DEV-2233

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ on:
       - master
 jobs:
   release_bin:
+    if: "${{ startsWith(github.event.head_commit.message, 'chore(server): release v') == true }}"
```
Comment (laurentlp) [inline .github/workflows/release.yml:8]: It basically means that we only create a new release when the last commit message on master starts with `chore(server): release v`. Note that only the Create Release Pull Request workflow creates those commits.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -16,16 +17,13 @@ jobs:
           node-version-file: '.nvmrc'
           cache: 'yarn'
 
-      - name: Git Unshallow
-        run: git fetch --prune --unshallow
```
Comment (laurentlp) [inline .github/workflows/release.yml:20]: Unnecessary since the `Get Release Details` already does so.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -36,13 +34,11 @@ jobs:
           echo "Latest Tag: ${{ steps.release_details.outputs.latest_tag }}"
 
       - name: Build and Package
-        if: ${{ steps.release_details.outputs.is_new_release == 'true' }}
         run: |
           yarn build
           yarn package
 
       - name: Release
-        if: ${{ steps.release_details.outputs.is_new_release == 'true' }}
```
Comment (laurentlp) [inline .github/workflows/release.yml:45]: Those two ifs were replaced by the one at the beginning of the file.

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@botpress/messaging",
-  "version": "1.1.4",
+  "version": "1.1.5",
```
Comment (laurentlp) [inline package.json:3]: I bumped the version here since we are working on v1.1.5

**Comment by laurentlp** [comment]
@samuelmasse I tested those actions locally using `act` and most of them were inspired by actions on the main repo. Also, we will be able to remove the release script from the repo once we know this is stable.

**Review by samuelmasse — APPROVED** [review]

**Pull request merged**